mteb/models/google_models.py

```diff
@@ -70,17 +108,54 @@ def encode(
         prompt_type: PromptType | None = None,
         **kwargs: Any,
     ) -> np.ndarray:
-        google_task_type = self.get_prompt_name(
-            self.model_prompts, task_name, prompt_type
+        prompt_name = self.get_prompt_name(self.model_prompts, task_name, prompt_type)
+        google_task_type = self.model_prompts.get(prompt_name)
+
+        show_progress_bar = (
+            False
+            if "show_progress_bar" not in kwargs
+            else kwargs.pop("show_progress_bar")
+        )
+
+        return self._embed(
+            sentences,
+            google_task_type=google_task_type,
+            show_progress_bar=show_progress_bar,
         )
-        return self._embed(sentences, google_task_type=google_task_type)
 
 
-name = "text-embedding-004"
-google_emb_004 = ModelMeta(
+google_text_emb_004 = ModelMeta(
+    loader=partial(
+        GoogleTextEmbeddingModel,
+        model_name="text-embedding-004",
+        model_prompts={
```

Comment:
Maybe move the prompts to a dictionary and pass the same dictionary to the loaders